hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -242,6 +249,11 @@ public HoodieFileGroupReaderIterator<T> getClosableIterator() {
     return new HoodieFileGroupReaderIterator<>(this);
   }
 
+  public static RecordMergeMode getRecordMergeMode(Properties props) {
+    String mergeMode = getStringWithAltKeys(props, HoodieCommonConfig.RECORD_MERGE_MODE, true).toUpperCase();
```

Comment:
note: Setting `useDefaultValue` to true as many tests don't set record merge mode.

Reply:
Sounds good.  The record merge mode is required to dictate the merging behavior in release 1.x, playing the same role as the payload class config in the release 0.x.  During table upgrade, we need to infer the record merge mode based on the payload class so it's correctly set.  HUDI-7847 to track the work.

Reply:
Right now, since there is only placeholder upgrade and downgrade methods from between table version 6 and 8, I added the inference of record merge mode inside `HoodieTableMetaClient$PropertyBuilder#build`.